man/msql2mysql.1
'\" t
.\"     Title: \fBmsql2mysql\fR
.\"    Author: [FIXME: author] [see http://docbook.sf.net/el/author]
.\" Generator: DocBook XSL Stylesheets v1.75.2 <http://docbook.sf.net/>
.\"      Date: 04/06/2010
.\"    Manual: MySQL Database System
.\"    Source: MySQL 5.1
.\"  Language: English
.\"
.TH "\FBMSQL2MYSQL\FR" "1" "04/06/2010" "MySQL 5\&.1" "MySQL Database System"
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.\" msql2mysql
.SH "NAME"
msql2mysql \- convert mSQL programs for use with MySQL
.SH "SYNOPSIS"
.HP \w'\fBmsql2mysql\fR\fB\fIC\-source\-file\fR\fR\fB\ \&.\&.\&.\fR\ 'u
\fBmsql2mysql\fR\fB\fIC\-source\-file\fR\fR\fB \&.\&.\&.\fR
.SH "DESCRIPTION"
.PP
Initially, the MySQL C API was developed to be very similar to that for the mSQL database system\&. Because of this, mSQL programs often can be converted relatively easily for use with MySQL by changing the names of the C API functions\&.
.PP
The
\fBmsql2mysql\fR
utility performs the conversion of mSQL C API function calls to their MySQL equivalents\&.
\fBmsql2mysql\fR
converts the input file in place, so make a copy of the original before converting it\&. For example, use
\fBmsql2mysql\fR
like this:
.sp
.if n \{\
.RS 4
.\}
.nf
shell> \fBcp client\-prog\&.c client\-prog\&.c\&.orig\fR
shell> \fBmsql2mysql client\-prog\&.c\fR
client\-prog\&.c converted
.fi
.if n \{\
.RE
.\}
.PP
Then examine
client\-prog\&.c
and make any post\-conversion revisions that may be necessary\&.
.PP
\fBmsql2mysql\fR
uses the
\fBreplace\fR
utility to make the function name substitutions\&. See
\fBreplace\fR(1)\&.
.SH "COPYRIGHT"
.br
.PP
Copyright 2007-2008 MySQL AB, 2008-2010 Sun Microsystems, Inc.
.PP
This documentation is free software; you can redistribute it and/or modify it only under the terms of the GNU General Public License as published by the Free Software Foundation; version 2 of the License.
.PP
This documentation is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
.PP
You should have received a copy of the GNU General Public License along with the program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA or see http://www.gnu.org/licenses/.
.sp
.SH "SEE ALSO"
For more information, please refer to the MySQL Reference Manual,
which may already be installed locally and which is also available
online at http://dev.mysql.com/doc/.
.SH AUTHOR
Sun Microsystems, Inc. (http://www.mysql.com/).


ChangeLog
2010-03-18	The InnoDB Team

	* CMakeLists.txt:
	Fix Bug#52102 InnoDB Plugin shows performance drop compared to
	InnoDB (Windows)

2010-03-18	The InnoDB Team

	* buf0buf.ic:
	When comparing the time of the first access to a block against
	innodb_old_blocks_time, use 32-bit arithmetics. The comparison was
	incorrect on 64-bit systems.

2010-03-11	The InnoDB Team

	* buf0buf.h, buf0buf.ic:
	Fix and clarify the latching of some buf_block_t members.
	Note that check_index_page_at_flush is not protected by any mutex.
	Note and assert that lock_hash_val is protected by the rw-latch.

2010-03-10	The InnoDB Team

	* trx/trx0sys.c:
	Fix Bug#51653 outdated reference to set-variable

2010-03-10	The InnoDB Team

	* handler/ha_innodb.cc, mysql-test/innodb_bug21704.result,
	mysql-test/innodb_bug47621.result, mysql-test/innodb_bug47621.test:
	Fix Bug#47621 MySQL and InnoDB data dictionaries will become out of
	sync when renaming columns

2010-03-10	The InnoDB Team

	* handler/ha_innodb.cc:
	Fix Bug#51356 Many Valgrind errors in error messages
	with concurrent DDL

2010-03-10	The InnoDB Team

	* handler/ha_innodb.cc, handler/handler0alter.cc,
	mysql-test/innodb_bug51378.result, mysql-test/innodb_bug51378.test:
	Fix Bug#51378 Init 'ref_length' to correct value, in case an out
	of bound MySQL primary_key

2010-03-10	The InnoDB Team

	* log/log0recv.c:
	Remove a bogus assertion about page numbers exceeding 0x90000000
	in the redo log. Abort when encountering a corrupted redo log
	record, unless innodb_force_recovery is set.

2010-03-09	The InnoDB Team

	* handler/ha_innodb.cc:
	Make SHOW ENGINE INNODB MUTEX STATUS display SUM(os_waits)
	for the buffer pool block mutexes and locks.

2010-03-08	The InnoDB Team

	* fil/fil0fil.c:
	Fix ALTER TABLE ... IMPORT TABLESPACE of compressed tables.

2010-03-03	The InnoDB Team

	* handler/handler0alter.cc, innodb-index.result, innodb-index.test,
	innodb.result, innodb.test:
	Disallow a duplicate index name when creating an index.

2010-02-11	The InnoDB Team

	* include/mem0mem.h, include/mem0mem.ic, mem/mem0mem.c:
	Fix Bug#49535 Available memory check slows down crash
	recovery tens of times

2010-02-09	The InnoDB Team

	* buf/buf0buf.c:
	Fix Bug#38901 InnoDB logs error repeatedly when trying to load
	page into buffer pool

2010-02-09	The InnoDB Team

	* srv/srv0srv.c:
	Let the master thread sleep if the amount of work to be done is
	calibrated as taking less than a second.

2010-02-04	The InnoDB Team

	* btr/btr0btr.c, btr/btr0cur.c, btr/btr0pcur.c, buf/buf0buf.c,
	include/btr0btr.h, include/btr0cur.h, include/btr0pcur.h,
	include/btr0pcur.ic, include/buf0buf.h, row/row0ins.c, row/row0sel.c:
	Pass the file name and line number of the caller of the
	b-tree cursor functions to the buffer pool requests, in order
	to make the latch diagnostics more accurate.

2010-02-03	The InnoDB Team

	* lock/lock0lock.c:
	Fix Bug#49001 SHOW INNODB STATUS deadlock info incorrect
	when deadlock detection aborts

2010-02-03	The InnoDB Team

	* buf/buf0lru.c:
	Fix Bug#35077 Very slow DROP TABLE (ALTER TABLE, OPTIMIZE TABLE)
	on compressed tables

2010-02-03	The InnoDB Team

	* handler/ha_innodb.cc, include/row0mysql.h, row/row0mysql.c:
	Clean up CHECK TABLE error handling.

2010-02-01	The InnoDB Team

	* handler/ha_innodb.cc, mysql-test/innodb-autoinc.test,
	mysql-test/innodb-autoinc.result,
	mysql-test/innodb-autoinc-44030.test,
	mysql-test/innodb-autoinc-44030.result:
	Fix Bug#49497 Error 1467 (ER_AUTOINC_READ_FAILED) on inserting
	a negative value

2010-01-27	The InnoDB Team

	* include/row0mysql.h, log/log0recv.c, row/row0mysql.c:
	Drop temporary tables at startup.
	This addresses the third aspect of
	Bug#41609 Crash recovery does not work for InnoDB temporary tables.

2010-01-21	The InnoDB Team

	* buf/buf0buf.c:
	Do not merge buffered inserts to compressed pages before
	the redo log has been applied in crash recovery.

2010-01-13	The InnoDB Team

	* row/row0sel.c:
	On the READ UNCOMMITTED isolation level, do not attempt to access
	a clustered index record that has been marked for deletion. The
	built-in InnoDB in MySQL 5.1 and earlier would attempt to retrieve
	a previous version of the record in this case.

2010-01-13	The InnoDB Team

	* buf/buf0buf.c:
	When disabling the adaptive hash index, check the block state
	before checking block->is_hashed, because the latter may be
	uninitialized right after server startup.

2010-01-12	The InnoDB Team

	* handler/ha_innodb.cc, handler/ha_innodb.h:
	Fix Bug#46193 crash when accessing tables after enabling
	innodb_force_recovery option

2010-01-12	The InnoDB Team

	* row/row0mysql.c:
	Fix Bug#49238 Creating/Dropping a temporary table while at 1023
	transactions will cause assert.

2009-12-02	The InnoDB Team

	* srv/srv0start.c:
	Display the zlib version number at startup.
	InnoDB compressed tables use zlib, and the implementation depends
	on the zlib function compressBound(), whose definition was slightly
	changed in zlib version 1.2.3.1 in 2006. MySQL bundles zlib 1.2.3
	from 2005, but some installations use a more recent zlib.

2009-11-30	The InnoDB Team

	* dict/dict0crea.c, dict/dict0mem.c, dict/dict0load.c,
	dict/dict0boot.c, fil/fil0fil.c, handler/ha_innodb.cc,
	include/dict0mem.h, row/row0mysql.c:
	Fix the bogus warning messages for non-existing temporary
	tables that were reported in
	Bug#41609 Crash recovery does not work for InnoDB temporary tables.
	The actual crash recovery bug was corrected on 2009-04-29.

2009-11-27	The InnoDB Team

	InnoDB Plugin 1.0.6 released

2009-11-20	The InnoDB Team

	* handler/ha_innodb.cc:
	Add a workaround to prevent a crash due to Bug#45961 DDL on
	partitioned innodb tables leaves data dictionary in an inconsistent
	state

2009-11-19	The InnoDB Team

	* btr/btr0btr.c:
	Fix Bug#48469 when innodb tablespace is configured too small, crash
	and corruption!

2009-11-19	The InnoDB Team

	* data/data0type.c:
	Fix Bug#48526 Data type for float and double is incorrectly reported
	in InnoDB table monitor


scripts/msql2mysql.sh
#!/bin/sh
# Copyright (c) 1979, 2014, Oracle and/or its affiliates. All rights reserved.
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program; see the file COPYING. If not, write to the
# Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston
# MA  02110-1301  USA.

echo "Warning: $0 is deprecated and will be removed in a future version.\n" 1>&2;

@bindir@/replace msqlConnect mysql_connect msqlListDBs  mysql_list_dbs msqlNumRows mysql_num_rows msqlFetchRow mysql_fetch_row msqlFetchField mysql_fetch_field msqlFreeResult mysql_free_result msqlListFields mysql_list_fields msqlListTables mysql_list_tables msqlErrMsg 'mysql_error(mysql)' msqlStoreResult mysql_store_result msqlQuery mysql_query msqlField mysql_field msqlSelect mysql_select msqlSelectDB mysql_select_db msqlNumFields mysql_num_fields msqlClose mysql_close msqlDataSeek mysql_data_seek m_field MYSQL_FIELD m_result MYSQL_RES m_row MYSQL_ROW msql mysql mSQL mySQL MSQL MYSQL msqlCreateDB mysql_create_db msqlDropDB mysql_drop_db msqlFieldSeek mysql_field_seek -- $*
